### DISCUSSION TITLE
HowTo Install Redmine in a sub-URI

### DISCUSSION THREAD

**Discussion by rolkey** [docker-library/redmine#359] [General]
I will put redmine docker in sub-URI，but i fail with： Page not found

```conf
    location /rmi/ {
        proxy_pass http://localhost:3000;  # 注意这里没有路径
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        # 保留原始路径信息
        proxy_set_header X-Original-URI $request_uri;
    }
```

**Comment by pianotako** [comment]
If you are using a Redmine Docker image version 5.0 or above, the frontend utilizes Puma.
According to the instructions on Redmine webpage, you need to make two modifications:
https://www.redmine.org/projects/redmine/wiki/HowTo_Install_Redmine_in_a_sub-URI#With-puma

First, you need to add the following environment variable to your docker-compose.yml 
```yml
    environment:
      RAILS_RELATIVE_URL_ROOT: "/rmi"
```
Second, you need to modify the `config.ru` file inside the container as follows:
file path => `/usr/src/redmine/config.ru`
patch
```diff
# This file is used by Rack-based servers to start the application.

require_relative 'config/environment'
-run Rails.application
+map ENV['RAILS_RELATIVE_URL_ROOT'] || '/' do
+    run RedmineApp::Application
+end
```
You can add this patch to your build process to create the correct image with the modified config.ru, or you can apply this patch manually after the container is brought up.
